### PR TITLE
test: add CIS 1.1.3 global admin count tests

### PR DESCRIPTION
**Pull request by karansoni15-create**
## Summary

- Added automated Rego tests for CIS Microsoft 365 Foundations control 1.1.3.
- Covered the compliant range of 2–4 Global Administrators.
- Added non-compliant tests for fewer than 2 and more than 4 Global Administrators.
- Added edge-case coverage for empty and missing `global_admins` data.
- Added validation of the policy result structure and recommended thresholds.

## Testing

Validated the complete OPA policy test suite using OPA 1.13.1.

Result:

PASS: 41/41

All CIS 1.1.3 tests passed successfully.

## Control

CIS Microsoft 365 Foundations v6.0.0

Control: 1.1.3 — Ensure that between two and four global admins are designated.

### CODE REVIEW

**Comment by akshitpatel1732** [comment]
@codex review

**Review by akshitpatel1732 — CHANGES_REQUESTED** [review]
Hi @karansoni15-create, thanks for your contributions around this test. Could you please look into the Codex suggestion to ensure this test is actually used when necessary? 

Once this issue is resolved, I'll review this PR again.

**Comment by akshitpatel1732** [comment]
Hi @karansoni15-create, is this PR still being worked on? 

I noticed that the issue I highlighted is still awaiting resolution. In addition, this PR now includes an additional file beyond the scope of this PR. Could you please remove the unrelated file as well? 
Thanks.

Note: The code freeze for this trimester (T2 2026) is 14/09/2026.

File: engine/tests/test_cis_1_1_3_global_admin_count.rego [hunk 1]
```diff
@@ -0,0 +1,158 @@
+package cis.microsoft_365_foundations.v6_0_0.test_control_1_1_3
+
+import rego.v1
```
[the remark this answers is not in the document]
Comment (karansoni15-create) [reply engine/tests/test_cis_1_1_3_global_admin_count.rego]: Addressed. I removed the standalone Rego test file and moved the added scenarios into the existing Compliance Engine behavioural fixture framework. Those fixtures are executed by the existing Engine CI `compliance-verification` job via `python -m scripts.validate_engine`, so regressions in 1.1.3 now fail the CI gate. I also removed the unrelated workflow changes from this PR.

**Comment by karansoni15-create** [comment]
I’ve addressed the remaining PR-side issues without changing the production policy or metadata. The 1.1.3 scenarios are now in the existing compliance behavioural fixture framework, the JSON fixtures have been formatted to satisfy the repository linter, and the OPA/compliance verification is passing. A fresh review has already been requested.
